cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/common/KafkaCruiseControlThreadFactory.java

```diff
@@ -17,9 +17,11 @@
   private final AtomicInteger _id = new AtomicInteger(0);
   private final Logger _logger;
 
-  public KafkaCruiseControlThreadFactory(String name,
-                                         boolean daemon,
-                                         Logger logger) {
+  public KafkaCruiseControlThreadFactory(String name) {
+    this(name, true, null);
+  }
+
+  public KafkaCruiseControlThreadFactory(String name, boolean daemon, Logger logger) {
```

Comment:
Just curious: in what situation(s) CC needs to use a non-daemon thread?

Reply:
A daemon thread provides a general service in the background as long as the program is running. It does not prevent the JVM from exiting when the program finishes but the thread is still running.

For example, when CC is executing a proposal, we may not want to let JVM exit until we make sure that the execution cleanup is handled properly -- i.e. stopped/cancelled tasks. Hence, if execution is still in progress, it might take slightly longer to shutdown the executor. That's why we have a non-daemon proposal executor.

Reply:
Makes sense